forch/forch_metrics.py

```diff
@@ -85,6 +107,31 @@ def get_metrics(self, path, params):
         """Return metric list in printable form"""
         return generate_latest(self._reg).decode('utf-8')
 
+    def _get_path_metrics(self, path):
+        url = self._metric_pages.get(path)
+        try:
+            data = requests.get(url)
+        except requests.exceptions.RequestException as e:
+            return "Error retrieving data from url %s: %s" % (url, str(e))
+        return data.content.decode('utf-8')
+
     def _show_error(self, error, path, params):
         """Display errors"""
         return f"Error creating varz interface: {str(error)}"
+
+    def _get_url(self, server, port):
+        return 'http://' + str(server) + ':' + str(port)
+
+    def _register_metric_page(self, path, server, port):
+        self._metric_pages[path] = self._get_url(server, port)
+
+    def _register_metric_pages(self):
+        self._register_metric_page('faucet', LOCALHOST,
```

Comment:
this should all be dynamicallky pulled from config, not hard-coded

Reply:
Addressed.